trying out solution for format_shapefile

njtierney opened the pull request:
https://github.com/ropensci/targets/discussion…

njtierney committed:
trying out solution in https://github.com/ropensci/targets/discussions/1240

brownag commented on _targets.R:
```diff
@@ -16,14 +16,14 @@ tar_plan(
   country_codes = country_codes(query = "Australia"),
 
   tar_target(
     example_gadm,
```
```suggestion
    example_gadm.zip,
```

The main trick to using /vsizip/ is the target name (and thus the file the target is stored in) has to end with ".zip". 

This naming affects both writing and reading, mainly because the code as written expects a .shp file in the project directory to ZIP and cleanup after. The code could be altered to work with a subdirectory, but you still need the .zip in the resulting target object stored for subsequent reading (or GDAL will not recognize it as a zip file source)

On first run of your code GDAL creates `"example_gadm"` folder with `"example_gadm.shp"` within it, resulting in `Layer 'example_gadm' already exists (GDAL error 1)` after running the pipeline at least once.

brownag commented on R/format_geoparquet.R:
```diff
@@ -0,0 +1,16 @@
+      dplyr::mutate(dplyr::across(wk::is_handleable, wk::as_wkb))
+  },
+  write = function(object, path) {
+    arrow::write_parquet(object, path)
```
I needed to convert the SpatVector -> data.frame for this to work (EDIT: I left out `geom="WKT"` to preserve geometry). 

```suggestion
    arrow::write_parquet(as.data.frame(object, geom="WKT"), path)
```

brownag commented on R/format_geoparquet.R:
```diff
@@ -0,0 +1,16 @@
+    arrow::write_parquet(object, path)
+  },
+  marshal = function(object) {
+    arrow::as_arrow_table(object)
```
Possibly here too? EDIT: as above

```suggestion
    arrow::as_arrow_table(as.data.frame(object, geom = "WKT"))
```

brownag commented on R/format_geoparquet.R:
```diff
@@ -0,0 +1,16 @@
+format_geoparquet <- tar_format(
+  read = function(path) {
+    arrow::read_parquet(path) |>
+      dplyr::mutate(dplyr::across(wk::is_handleable, wk::as_wkb))
```
In order to roundtrip a SpatVector with geometry from parquet, need `vect(geom="geometry")`
```suggestion
      dplyr::mutate(dplyr::across(wk::is_handleable, wk::as_wkb)) |>
      terra::vect(geom = "geometry")
```

njtierney committed:
Use .zip suffix on target names

From @brownag:

* The main trick to using /vsizip/ is the target name (and thus the file the target is stored in) has to end with ".zip".
* This naming affects both writing and reading, mainly because the code as written expects a .shp file in the project directory to ZIP and cleanup after. The code could be altered to work with a subdirectory, but you still need the .zip in the resulting target object stored for subsequent reading (or GDAL will not recognize it as a zip file source)
* On first run of your code GDAL creates "example_gadm" folder with "example_gadm.shp" within it, resulting in Layer 'example_gadm' already exists (GDAL error 1) after running the pipeline at least once.

Co-authored-by: Andrew Gene Brown <andrew.g.brown@usda.gov>

njtierney committed:
Update R/format_geoparquet.R

From @brownag:

* I needed to convert the SpatVector -> data.frame for this to work (EDIT: I left out geom="WKT" to preserve geometry).

Co-authored-by: Andrew Gene Brown <andrew.g.brown@usda.gov>

njtierney committed:
Update R/format_geoparquet.R

from @brownag:

* update geom = "WKT"

Co-authored-by: Andrew Gene Brown <andrew.g.brown@usda.gov>

njtierney committed:
Update R/format_geoparquet.R

from @brownag:

In order to roundtrip a SpatVector with geometry from parquet, need vect(geom="geometry")

Co-authored-by: Andrew Gene Brown <andrew.g.brown@usda.gov>

njtierney committed:
Misc changes:

* gitignore data folder
* remove comment on tar_plan
* update meta/meta

njtierney commented:
That all works for me, thanks so much!
